Readability.js

```diff
@@ -27,6 +27,10 @@
  * @param {Object}       options The options object.
  */
 function Readability(uri, doc, options) {
+  if (arguments.length === 1) {
```

Comment:
Ah, hm. The test changes made me realize this way of checking arguments means we can't omit the URL when passing options... That seems unfortunate.

Let's be a little bit more forward and change the function signature, and then use duck typing to figure out what people are passing, so we get:

```js
function Readability(doc, options) {
  // In some older versions, people passed a URI object as the first argument. Cope:
  if (doc && !doc.documentElement && doc.spec) {
    doc = options;
    options = arguments[2];
  }
```

Please also update the doc comment above the constructor definition to omit the `uri` bit. :-)

Then I think we can remove the explicit `null` being passed from the callsites in `test/`.